internal/cli/plugin/plugin.go

```diff
@@ -69,7 +78,7 @@ func (opts *Opts) findPluginWithName(name string) (*plugin.Plugin, error) {
 }
 
 func RegisterCommands(rootCmd *cobra.Command) {
-	plugins := plugin.GetAllValidPlugins(rootCmd.Commands())
+	plugins := plugin.GetAllValidPlugins(createExistingCommandsMap(rootCmd.Commands()))
```

Comment:
Why don't you register your first class plugins in this function so you don't have to rely on annotations at all?

Reply:
done